apl-core/src/test/java/com/apollocurrency/aplwallet/apl/core/service/state/currency/CurrencyServiceTest.java

```diff
@@ -148,7 +155,7 @@ void setUp() {
         service = new CurrencyServiceImpl(currencySupplyTable, currencyTable, currencyMintTable, monetaryCurrencyMintingService, blockChainInfoService,
             accountService, accountCurrencyService, currencyExchangeOfferFacade, currencyFounderService,
             exchangeService, currencyTransferService, shufflingService, blockchainConfig,
-            transactionValidationHelper, fullTextSearchUpdater, fullTextSearchService);
+            transactionValidationHelper, fullTextSearchUpdater, fullTextOperationDataEvent, fullTextSearchService);
```

Comment:
Maybe it would be better to use just fullTextSearchService to index new rows instead of using events? Seems like it adds additional redundant dependencies

Reply:
That is not quite correct.
1. Either fullTextSearchUpdater or fullTextOperationDataEvent is used for altering data. fullTextSearchUpdater uses FullTextSearchEngine for that.
2. FullTextSearchService is used for searching and it has start up/drop down logic.

Reply:
Got you